src/server/controller/plugin.ts

```diff
@@ -9,48 +8,62 @@ export default class Controller {
     this.db = new PostgresGateway();
   }
 
-  public async write(body: Trade): Promise<{}> {
+  public async write(
+    body: Trade
+  ): Promise<{
+    rows: Array<{ id: string }>;
+  }> {
     try {
       const id = await this.db.write(body);
       return id;
-    } catch (error) {
-      throw error;
+    } catch (err) {
+      this.db.logger.log("error", "Error:", err);
+
+      throw err;
     }
   }
 
-  public async read(id: string): Promise<Trade> {
+  public async read(id: string): Promise<object> {
```

Comment:
**Comment**: This is going to sound like a nitpick, but it isn't. We should be (ideally) returning `Promise<Trade>`. Think of it this way: whoever calls Controller.read() will now get something of type `object` back, which indicates a stark loss of type precision. Because we cannot control who will consume our function and how, we must be careful  and responsible to not infect the type purity of our system by introducing programmatic entropy.

Reply:
Specified type.

Reply:
Additionally, response on server side now have specified type as well.